libraries/GCS_MAVLink/MissionItemProtocol_Fence.cpp

```diff
@@ -226,7 +226,7 @@ MAV_MISSION_RESULT MissionItemProtocol_Fence::allocate_receive_resources(const u
 MAV_MISSION_RESULT MissionItemProtocol_Fence::allocate_update_resources()
 {
     const uint16_t _item_count = _fence.polyfence().num_stored_items();
-    _updated_mask = new uint8_t[(_item_count+7/8)];
+    _updated_mask = new uint8_t[(_item_count + static_cast<uint8_t>(7.0/8.0))];
```

Comment:
```suggestion
    _updated_mask = new uint8_t[(_item_count + 7)/8.0)];
```

This is attempting to round up.